pubsub/tests/system.py

```diff
@@ -183,6 +184,218 @@ def test_subscribe_to_messages_async_callbacks(
+
+    # there might be other subscriptions in the project, thus do a "is subset" check
+    assert set(subscription_paths) <= subscriptions
+
+
+def test_listing_topic_subscriptions(publisher, subscriber, project, cleanup):
+    # create topics
+    topic_paths = [
+        publisher.topic_path(project, "topic-1" + unique_resource_id(".")),
+        publisher.topic_path(project, "topic-2" + unique_resource_id(".")),
+    ]
+    for topic in topic_paths:
+        cleanup.append((publisher.delete_topic, topic))
+        publisher.create_topic(topic)
+
+    # create subscriptions
+    subscription_paths = [
+        subscriber.subscription_path(
+            project, "sub-{}".format(i) + unique_resource_id(".")
+        )
+        for i in range(1, 4)
+    ]
+    for i, subscription in enumerate(subscription_paths):
+        topic = topic_paths[i % 2]
+        cleanup.append((subscriber.delete_subscription, subscription))
+        subscriber.create_subscription(subscription, topic)
+
+    # retrieve subscriptions and check that the list matches the expected
+    subscriptions = publisher.list_topic_subscriptions(topic_paths[0])
+    subscriptions = set(subscriptions)
+
+    assert subscriptions == {subscription_paths[0], subscription_paths[2]}
+
+
+def test_using_snapshots(
+    publisher, subscriber, project, topic_path, subscription_path, cleanup
+):
+    snapshot_path = subscriber.snapshot_path(project, "my-snapshot")
+
+    # Make sure the topic, subscription, and snapshot get deleted.
+    cleanup.append((publisher.delete_topic, topic_path))
+    cleanup.append((subscriber.delete_subscription, subscription_path))
+    cleanup.append((subscriber.delete_snapshot, snapshot_path))
+
+    # publish a message to a topic, pull and acknowledge it
+    publisher.create_topic(topic_path)
+    subscriber.create_subscription(subscription_path, topic_path)
+
+    publisher.publish(topic_path, b"Message 1")
+    response = subscriber.pull(subscription_path, max_messages=100)
+    assert len(response.received_messages) == 1
+    msg = response.received_messages[0]
+    subscriber.acknowledge(subscription_path, [msg.ack_id])
+
+    # repeat the same *after* creating a snapshot
+    subscriber.create_snapshot(snapshot_path, subscription_path)
+
+    publisher.publish(topic_path, b"Message 2")
+    response = subscriber.pull(subscription_path, max_messages=100)
+    msg = response.received_messages[0]
+    subscriber.acknowledge(subscription_path, [msg.ack_id])
+
+    # pulling again, there should be no messages received
+    response = subscriber.pull(
+        subscription_path, max_messages=100, return_immediately=True
+    )
+    assert len(response.received_messages) == 0
+
+    # after seeking to the snapshot, the second message should be received again
+    subscriber.seek(subscription_path, snapshot=snapshot_path)
+    response = subscriber.pull(subscription_path, max_messages=100)
+    assert len(response.received_messages) == 1
```

Comment:
This fails when run with other system tests, a message is not received (but seems to consistently pass when run in isolation). From the [snapshot docs](https://cloud.google.com/pubsub/docs/replay-qs):
> You can recover the message by seeking the subscription to the snapshot. Note that the message may not be delivered instantly: Pub/Sub does not guarantee message delivery latency. However, what is guaranteed is that the messages retained in the snapshot will be eventually delivered at least once

Since the message will only be _eventually_ delivered, this test is apparently flaky. Will have to think a way around this...

Reply:
Removed the test case, as the test cannot wait for the message to "eventually" arrive ... or not.